procrastinate/sql/migrations/03.02.00_01_add_retry_failed_job_procedure.sql

```diff
@@ -0,0 +1,78 @@
+-- Append new event type to reflect transition from failed -> todo
+ALTER TYPE procrastinate_job_event_type ADD VALUE 'retried' AFTER 'scheduled';
+
+-- Procedure to retry failed jobs
+CREATE FUNCTION procrastinate_retry_failed_job_v1(
+    job_id bigint,
+    new_priority integer,
+    new_queue_name character varying,
+    new_lock character varying
+) RETURNS void LANGUAGE plpgsql AS $$
+DECLARE
+    _job_id bigint;
+BEGIN
+    UPDATE procrastinate_jobs
+    SET status = 'todo'::procrastinate_job_status,
+        attempts = attempts + 1,
+        priority = COALESCE(new_priority, priority),
+        queue_name = COALESCE(new_queue_name, queue_name),
+        lock = COALESCE(new_lock, lock)
```

Comment:
I see you specifically removed the options that allow scheduling the retry in the future. I could imagine some use cases where it might be useful. Do you have an opinion on that ?

Reply:
Well, the function could accept a date without Django using it? That would still let people who use other integration options to make this possible?

That said, we can always do that in the future, it's just that I saw you specifically _removed_ those.